root/artist_credit/ArtistCreditIndex.js

```diff
@@ -0,0 +1,136 @@
+import DescriptiveLink
+  from '../static/scripts/common/components/DescriptiveLink';
+import EntityLink
+  from '../static/scripts/common/components/EntityLink';
+import expand2text from '../static/scripts/common/i18n/expand2text';
+
+import ArtistCreditLayout from './ArtistCreditLayout';
+
+type Props = {
+  +$c: CatalystContextT,
+  +artistCredit: {+id: number} & ArtistCreditT,
+  +creditedEntities: {
+    +[entityType: string]: {
+      +count: number,
+      +entities: $ReadOnlyArray<{
+        +entity: CoreEntityT | TrackT,
+        +entity_id: number,
+      }>,
+    },
+  },
+};
+
+function buildSection(
+  props: Props,
+  entityType: string,
+  title: string,
+  seeAllMessage: $Call<typeof N_ln, string, string>,
+) {
+  const entities = props.creditedEntities[entityType];
+
+  if (!entities.count) {
+    return null;
+  }
+
+  const entityUrlFragment = ENTITIES[entityType].url;
+  const url = '/artist-credit/' + props.artistCredit.id +
+              '/' + entityUrlFragment;
+
+  return (
+    <React.Fragment key={entityType}>
+      <h3>{title}</h3>
+      <ul>
+        {entities.entities.map(entity => (
+          <li key={entity.entity_id}>
+            {entity.entity.entityType === 'track' ? (
+              <a href={`/track/${entity.entity.gid}`}>
+                {entity.entity.name}
+              </a>
+            ) : <EntityLink entity={entity.entity} />}
+          </li>
+        ))}
+        {entities.count > entities.entities.length ? (
+          <li key="see-all">
+            <em>
+              <a href={url}>
+                {expand2text(
+                  seeAllMessage(entities.count),
+                  {num: entities.count},
+                )}
+              </a>
+            </em>
+          </li>
+        ) : null}
+      </ul>
+    </React.Fragment>
+  );
+}
+
+const ArtistCreditIndex = (
+  props: Props,
+): React.Element<typeof ArtistCreditLayout> => (
+  <ArtistCreditLayout
+    $c={props.$c}
+    artistCredit={props.artistCredit}
+    page=""
+  >
+    <p>
+      {l('This artist credit is composed of the following artists:')}
+    </p>
+    <ul>
+      {props.artistCredit.names.map((name, index) => (
+        <li key={'name-' + index}>
+          <DescriptiveLink entity={name.artist} />
+          {name.artist.name === name.name ? null : (
+            <>
+              {' '}
+              {exp.l(
+                'credited as “{credit}”',
+                {credit: name.name},
+              )}
+            </>
+          )}
+        </li>
+      ))}
+    </ul>
+    <h2>{l('Uses')}</h2>
+    {/*
+      * The below use N_ln so languages with non-Germanic pluralization
+      * rules (i.e., any that make number distinctions above the
+      * threshold where we'll actually show the string) can translate
+      * properly. However, the strings are the same in English because
+      * we do not make a distinction other than for 1, which will never
+      * show in this case.
+    */}
```

Comment:
We might want to consider moving these strings to constants.js later so they can be reused without duplicating the comment and everything. I guess it's fine for now since it's only copied to here and the tags place.